extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -28,15 +30,71 @@ class AmpStickyAd extends AMP.BaseElement {
   isLayoutSupported(layout) {
     return layout == Layout.NODISPLAY;
   }
-
-  /** @override */
+  /**
+   * @override
+   * @this {undefined}  // Make linter happy
+   */
   buildCallback() {
     /** @const @private {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
     if (!this.isExperimentOn_) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+    this.viewport_ = this.getViewport();
+    this.isDisplayed_ = false;
+    this.initialScrollTop_ = this.viewport_.getScrollTop();
```

Comment:
Sticky ad should definitely be built. But you can't do measurements in the `buildCallback` directly: it's a mutate context and thus measurements are not allowed. What more - the measurements most likely to be incorrect here.

`layoutCallback` won't be called by itself since the element is initially not displayed.

You can certainly, however, add listeners in build callback - that's cheap and not related to measurement. What happens in the listener is the business of the listener and who sends events.

Finally, `scrollHeight` changes completely independent of this component. We simply can't store it once and use it forever. Whether or not it's expensive - we have to experiment and estimate. Currently we don't have a system or an AMP event that would give us this value. We can certainly find a way to cache it somewhere and update less frequently if it's expensive. But I'd start with reading on scroll callback and seeing what happens in timelines.